### PR TITLE
{devel}[system/system] Saxon-HE v12.4 w/ Java 21

### PR DESCRIPTION
**Pull request by SimonPinches**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by SimonPinches** [comment]
This PR relies upon #19620.

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/eebf4d3feee633df1314b87ccfdba2be for a full test report.

File: easybuild/easyconfigs/s/Saxon-HE/Saxon-HE-12.4-Java-21.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    '%(namelower)s-%(version)s.jar', '%(namelower)s-test-%(version)s.jar', '%(namelower)s-xqj-%(version)s.jar'
+]
+
+sanity_check_paths = {
+    'files': local_jarfiles,
+    'dirs': [],
+}
+
+modextrapaths = {'CLASSPATH': local_jarfiles}
+
```
Comment (Micket) [inline easybuild/easyconfigs/s/Saxon-HE/Saxon-HE-12.4-Java-21.eb:28]: This is added automatically now, and we get warnings if we keep this in. 
```suggestion
```

Comment (Micket) [inline easybuild/easyconfigs/s/Saxon-HE/Saxon-HE-12.4-Java-21.eb:28]:
```suggestion

```
done automatically, skip this to avoid warnings

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/1348d573855d2d2ceb7cec6f388b461a for a full test report.

**Comment by Micket** [comment]
@boegelbot please test @ jsc-zen3

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/b0bf8a6d7e610beb12cd69ac55d4a307 for a full test report.

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
Going in, thanks @SimonPinches!

**Pull request merged**